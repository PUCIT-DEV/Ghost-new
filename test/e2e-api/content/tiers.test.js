const {agentProvider, fixtureManager, matchers} = require('../../utils/e2e-framework');

describe('Tiers Content API', function () {
    let agent;

    before(async function () {
        agent = await agentProvider.getContentAPIAgent();
        await fixtureManager.init('members', 'api_keys');
        agent.authenticate();
    });

    it('Can request tiers', async function () {
        await agent.get('/tiers/')
            .expectStatus(200)
            .matchHeaderSnapshot({
                etag: matchers.anyEtag
            })
            .matchBodySnapshot({
                tiers: Array(2).fill({
                    id: matchers.anyObjectId,
                    created_at: matchers.anyISODate,
                    updated_at: matchers.anyISODate
                })
            });
    });
});
