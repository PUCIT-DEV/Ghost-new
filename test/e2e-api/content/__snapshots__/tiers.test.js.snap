// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tiers Content API Can request only active tiers 1: [body] 1`] = `
Object {
  "meta": Object {
    "pagination": Object {
      "limit": 15,
      "next": null,
      "page": 1,
      "pages": 1,
      "prev": null,
      "total": 2,
    },
  },
  "tiers": Array [
    Object {
      "active": true,
      "benefits": Array [],
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "description": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "name": "Free",
      "slug": "free",
      "type": "free",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "visibility": "public",
      "welcome_page_url": "/welcome-free",
    },
    Object {
      "active": true,
      "benefits": Array [],
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "currency": "GBP",
      "description": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "monthly_price": 1200,
      "name": "Default Product",
      "slug": "default-product",
      "type": "paid",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}/,
      "visibility": "public",
      "welcome_page_url": "/welcome-paid",
      "yearly_price": 12000,
    },
  ],
}
`;

exports[`Tiers Content API Can request only active tiers 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "*",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "700",
  "content-type": "application/json; charset=utf-8",
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Encoding",
  "x-powered-by": "Express",
}
`;
